happypose/toolbox/utils/download.py

```diff
@@ -173,27 +180,135 @@ def main():
+
+
+    async def adownload(self, download_path, local_path, flags):
+
+        Path_download = Path(download_path)
+        if Path_download.name != local_path.name:
+            local_path = local_path / Path_download.name
+        
+        if not flags.flags_managing(Path_download.name): #if the dl_path is --excluded
+            return
+
+        if not download_path.endswith("/") and not httpx.head(download_path).is_redirect: #file
+            await self.download_file(download_path, local_path)
+        else:
+            if not download_path.endswith("/"): 
+                download_path += "/"
+            await self.download_dir(download_path, local_path, flags)
+ 
+
+    async def download_dir(self, download_path, local_path, flags):
+        try:
+            r = await self.client.get(download_path)
+        except httpx.PoolTimeout:
+            logger.error(f"Failed {download_path} with timeout")
+            return
+        if r.status_code != 200:
+            logger.error(f"Failed {download_path} with code {res.status_code}")
+            return
+        create_dir(local_path)
+        soup = BeautifulSoup(r.content, "html.parser")
+        logger.info(f"Copying {download_path} to {local_path}")
+
+        for link in soup.find_all('a')[5:]:
+            href: str = link.get('href')
+            if not flags.flags_managing(href):
+                continue
+            if href.endswith("/"):
+                self.create_task(self.download_dir(download_path + href, local_path / href,flags))
+            else:
+                self.create_task(self.download_file(download_path + href, local_path / href))
+
+
+    async def download_file(self, download_path, local_path):
+        try:
+            r = await self.client.get(download_path)
+        except httpx.PoolTimeout:
+            logger.error(f"Failed {download_path} with timeout")
+            return
+        if r.status_code != 200:
+            logger.error(f"Failed {download_path} with code {res.status_code}")
+            return
+        logger.info(f"Copying {download_path} to {local_path}")
+        create_dir(local_path.parent)
+        with open(str(local_path),'wb') as f:
+            f.write(r.content)
+
+
+def create_dir(path):
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+class Flags:
+    def __init__(self, flags: [str]):
+        #only '--exclude' were used before so this is the only flag currently usable
+        #if you need to use other flags, feel free to implement them here
+        self.exclude_set: set[str] = set()
+
+        skip = False
+
+        for (arg, nxt) in zip(flags, flags[1:]):
+            if skip:
+                skip = False
+            elif arg == "--exclude":
+                self.exclude_set.add(nxt)
+                skip = True
+
+    def flags_managing(flags, href):
+        for el in flags.exclude_set:
+            if re.fullmatch(el, href):
+                return False
+        return True
```

Comment:
It would be better to try to use [`argparse`](https://docs.python.org/3.11/library/argparse.html) for this I think